docs/technical/error-reporting.md

```diff
@@ -0,0 +1,71 @@
+# Error Reporting
+
+First we need to make the distinction between expected runtime _errors_ and
+_exceptions_. Unfortunately both are represented with the `Error` class, but
+they're conceptually different. Exceptions are fatal, errors are not.
+
+The story around exceptions is simpler so let's start there.
+
+## Exceptions
+
+An exception is an unexpected, fatal problem in the app itself. For example, our
+old friend `undefined is not a function`. This is a problem with the code itself
+which cannot be resolved at runtime. Our only option is to quit the app and
+relaunch.
+
+We handle uncaught exceptions by registering a [global listener](https://github.com/desktop/desktop/blob/fb4e73560127f491ccf5f59984a310481911f2b6/app/src/ui/index.tsx#L75).
+We report the exception to Central, tell the user that an unrecoverable error
+happened, and then quit and relaunch. End of story.
+
+## Errors
+
+Errors are a bit more involved. They are anything that can go wrong in the
+standard usage of the app. For example, if the internet's down or a git
+repository is in a funny state, we're gonna get some errors.
+
+Our error reporting flows through the `Dispatcher` like most everything in the
+app. [`postError`](https://github.com/desktop/desktop/blob/fb4e73560127f491ccf5f59984a310481911f2b6/app/src/lib/dispatcher/dispatcher.ts#L308)
+calls the [registered](https://github.com/desktop/desktop/blob/fb4e73560127f491ccf5f59984a310481911f2b6/app/src/lib/dispatcher/dispatcher.ts#L711)
+[error handlers](https://github.com/desktop/desktop/blob/fb4e73560127f491ccf5f59984a310481911f2b6/app/src/lib/dispatcher/error-handlers.ts),
+starting with the most recently registered. The error handlers have the chance
```

Comment:
>And what's a good example of why you'd change the error?

"change" was probably the wrong word to use. I tried some better words, lemme know how they hit you.